cms/static/js/views/utils/create_course_utils.js

```diff
@@ -1,57 +1,18 @@
 /**
  * Provides utilities for validating courses during creation, for both new courses and reruns.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
+        "use strict";
         return function (selectors, classes) {
-            var validateTotalCourseItemsLength, setNewCourseFieldInErr, hasInvalidRequiredFields,
-                createCourse, validateFilledFields, configureHandlers;
+            CreateUtilsFactory.call(this, selectors, classes);
```

Comment:
Just when I thought I was understanding.... Why only pass selectors and classes to call, and then define the rest of the things afterwards (keyFieldSelectors, create function, etc.)? I'm not following what is "special" about selectors and classes.

Yes, JavaScript is often surprising!